pkg/apis/core/validation/validation.go

```diff
@@ -3682,6 +3701,57 @@ func validateHostUsers(spec *core.PodSpec, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
+// validateFileKeyRefVolume validates that volumes referenced by FileKeyRef environment variables
+// are of type emptyDir. FileKeyRef requires emptyDir volumes to ensure proper file access.
+func validateFileKeyRefVolume(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Collect all volume names referenced by FileKeyRef environment variables
+	referencedVolumeNames := sets.Set[string]{}
+	for _, container := range spec.Containers {
```

Comment:
Can initContainers and ephemeralContainers use this feature?  I forget what the KEP says, but either way it should be clear when looking at this code that these were not "forgotten".